cpp/examples/ode_secir_contact_changes.cpp

```diff
@@ -51,12 +51,12 @@ int main()
     model_a.populations[{mio::AgeGroup(0), mio::osecir::InfectionState::InfectedSymptoms}] = nb_inf_t0;
     model_a.populations.set_difference_from_total({mio::AgeGroup(0), mio::osecir::InfectionState::Susceptible},
                                                   nb_total_t0);
-    mio::ContactMatrixGroup& contact_matrix_a = model_a.parameters.get<mio::osecir::ContactPatterns>();
+    mio::ContactMatrixGroup& contact_matrix_a = model_a.parameters.get<mio::osecir::ContactPatterns<ScalarType>>();
     contact_matrix_a[0]                       = mio::ContactMatrix(Eigen::MatrixXd::Constant(1, 1, cont_freq));
     // set probability of transmission and risk of infection to 1.
-    model_a.parameters.get<mio::osecir::TransmissionProbabilityOnContact>() = 1.0;
-    model_a.parameters.get<mio::osecir::RiskOfInfectionFromSymptomatic>()   = 1.0;
-    auto result_a = simulate_flows(t0, tmax, dt, model_a, integrator);
+    model_a.parameters.get<mio::osecir::TransmissionProbabilityOnContact<ScalarType>>() = 1.0;
+    model_a.parameters.get<mio::osecir::RiskOfInfectionFromSymptomatic<ScalarType>>()   = 1.0;
+    auto result_a = mio::simulate_flows<ScalarType>(t0, tmax, dt, model_a, integrator);
```

Comment:
This also works without namespace and template, if we specify the type of `integrator` above instead of using auto. I do not have a strong preference for either option.